modules/openapi-generator/src/main/resources/typescript-rxjs/apis.mustache

```diff
@@ -37,11 +37,11 @@ export class {{classname}} extends BaseAPI {
      * {{&summary}}
      {{/summary}}
      */
-    {{nickname}} = ({{#allParams.0}}requestParameters: {{operationIdCamelCase}}Request{{/allParams.0}}): Observable<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/returnType}}> => {
+    {{nickname}} = ({{#allParams.0}}{ {{#allParams}}{{paramName}}{{^-last}}, {{/-last}}{{/allParams}} }: {{operationIdCamelCase}}Request{{/allParams.0}}): Observable<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/returnType}}> => {
```

Comment:
i suggest not to unpack the parameters, since it will lead to many naming conflicts

Reply:
You're right, they may conflict with `query`, `headers`, `body` and `formData` now. 
Those could either be suffixed with something like `nickname` or while destructuring the params the ones that collide get renamed. 
I am going to try the latter since it seems cleaner.

Reply:
I got it working like this where params are checked for the keys `query`, `headers` and `formData` (`body` wasn't an issue).

The result for those looks like 
```diff
-    loginUser = ({ formData, query }: LoginUserRequest): Observable<string> => {
-        throwIfNullOrUndefined(formData, 'loginUser');
-        throwIfNullOrUndefined(query, 'loginUser');
+    loginUser = ({ formData: formDataAlias, query: queryAlias }: LoginUserRequest): Observable<string> => {
+        throwIfNullOrUndefined(formDataAlias, 'loginUser');
+        throwIfNullOrUndefined(queryAlias, 'loginUser');
 
         const query: HttpQuery = { // required parameters are used directly since they are already checked by throwIfNullOrUndefined
-            'formData': formData,
-            'query': query,
+            'formData': formDataAlias,
+            'query': queryAlias,
         };
```

```diff
-    findPetsByTags = ({ headers }: FindPetsByTagsRequest): Observable<Array<Pet>> => {
-        throwIfNullOrUndefined(headers, 'findPetsByTags');
+    findPetsByTags = ({ headers: headersAlias }: FindPetsByTagsRequest): Observable<Array<Pet>> => {
+        throwIfNullOrUndefined(headersAlias, 'findPetsByTags');

        const headers: HttpHeaders = {
            ...
        };

         const query: HttpQuery = { // required parameters are used directly since they are already checked by throwIfNullOrUndefined
-            'headers': headers.join(COLLECTION_FORMATS['csv']),
+            'headers': headersAlias.join(COLLECTION_FORMATS['csv']),
         };
```

To get there I extended `CodegenParameter` by `paramNameAlternative` and `paramNameOrAlternative` in order to keep templates simple.
Inheriting from `CodegenParameter` an creating a new class would be possible but this would probably be a first in the project and I wanted to avoid a copy constructor because of maintainablity.

So now my question are:
1) Is this acceptable or should I extend `CodegenParameter` locally in the generator?
2) Is extending the java code favored over having logic in templates? In this case the only other way would be to have `{{#paramNameAlternative}}{{paramNameAlternative}}{{/paramNameAlternative}}{{^paramNameAlternative}}{{paramName}}{{/paramNameAlternative}}` in like 12 places instead of `{{paramNameOrAlternative}}`.

Reply:
thanks for your improvements.
I still don't understand why such an effort has to be made instead of just using the current `requestParameters`

Reply:
The main effort is already done. 
The result is more performant code at a reduced file size and believe anyone does welcome this.
Perhaps @wing328 can answer the two questions from above?

Reply:
Please ping me via Slack to have a discussion on this in Jan 2020 when you've time.